File: lib/astropical.js
const fetch = require('node-fetch');
const pMemoize = require('p-memoize');

const BaseUrl = 'http://astropical.space/';

const fetchPlanetEphem = async function(lat, lon) {
  const url = BaseUrl + 'api-ephem.php?lat=' + lat + '&lon=' + lon;
  return fetch(url).then(function(response) {
    return response.json();
  });
};

const getPlanetEphem = pMemoize(fetchPlanetEphem);

module.exports = {
  getPlanetEphem,
};

File: app/services/astropical.py
import httpx
from cachetools import TTLCache, cached

_cache = TTLCache(maxsize=64, ttl=300)


@cached(cache=_cache)
def get_planet_ephem(lat: float, lon: float) -> dict:
    url = f"https://astropical.space/api-ephem.php?lat={lat}&lon={lon}"
    resp = httpx.get(url, timeout=10)
    resp.raise_for_status()
    return resp.json()
